docs/contents/basic_usage.md

```diff
@@ -116,6 +116,28 @@ cargo install --force dmenv
 ```
 
 
+## Upgrading a top-level dependency
+
+Let's say your project depends on `foolib`. Version `1.3` works fine, but
+`1.4` just came out and even if you don't need any new feature from `1.4`,
+you'd like to check wether your project is compatible.
+
+Assuming you already have a virtual environment containing `foolib 1.3`, you can can do so by running:
+
+```bash
+$ git status  # check that requirements.lock is clean
+$ source $(dmenv show:bin_path)/activate
+$ pip install --upgrade foolib
+```
+
+Most of the time, `pip install --upgrade foolib` will do the right thing:
+
+* it will keep other dependencies installed at their current version
+* it will install new dependencies of `foolib 1.4` if they are missing
+* if `foolib 1.4` contains a `bar >= 3.0` constraint and you have `bar == 2.0` in the virtualenv, `bar` will
+  be upgraded too.
+
+Just to be sure, you can then inspect the differences in the lock file by hand, and if they look OK, you can commit and push the new version of the lock.
```

Comment:
do we need to lock dmenv again afterward?

Reply:
Yes